src/OpenTelemetry.AutoInstrumentation.StartupHook/RulesEngine/DiagnosticSourceRule.cs

```diff
@@ -14,13 +14,59 @@
 // limitations under the License.
 // </copyright>
 
+using System.Diagnostics;
+using System.Reflection;
+using OpenTelemetry.AutoInstrumentation.Logging;
+
 namespace OpenTelemetry.AutoInstrumentation.RulesEngine;
 
 internal class DiagnosticSourceRule : Rule
 {
+    private static readonly IOtelLogger Logger = OtelLogging.GetLogger("StartupHook");
+
+    public DiagnosticSourceRule()
+    {
+        Name = "System.Diagnostics.DiagnosticSource Validator";
+        Description = "Ensure that the System.Diagnostics.DiagnosticSource version is not older than the version used by the Auto-Instrumentation";
+    }
+
     internal override bool Evaluate()
     {
-        // TODO: implement checking logic
+        string? diagnosticSourcePackageVersion = null;
+
+        try
+        {
+            // Look up for type with an assembly name, will load the library.
+            // The loaded version depends on the app's reference to the package.
+            var diagnosticSourceType = Type.GetType("System.Diagnostics.DiagnosticSource, System.Diagnostics.DiagnosticSource");
```

Comment:
This looks incorrect, namespace for `System.Diagnostics.DiagnosticSource` is `System.Diagnostics`. Could you please verify?